src/pudl/transform/ferc1.py

```diff
@@ -154,6 +154,27 @@ def transform_start(
         logger.info(f"{self.table_id.value}: Concatenating DBF + XBRL dataframes.")
         return pd.concat([processed_dbf, processed_xbrl]).reset_index(drop=True)
 
+    @cache_df(key="main")
+    def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Generic FERC1 main table transformer.
+
+        Params:
+            df: Pre-processed, concatenated XBRL and DBF data.
+
+        Returns:
+            A single transformed table concatenating multiple years of cleaned data
+            derived from the raw DBF and/or XBRL inputs.
+        """
+        df = (
+            self.normalize_strings(df)
+            .pipe(self.categorize_strings)
+            .pipe(self.convert_units)
+            .pipe(self.strip_non_numeric_values)
+            .pipe(self.nullify_outliers)
+            .pipe(self.drop_invalid_rows)
+        )
+        return df
```

Comment:
I suspect this is a premature generalization, but we'll see.

Reply:
it might be. but i think bc a lot of these guys do nothing if there are no params, it feels good. I used it in the purchased power table just now and that table is verrry different.